for-loops.md

```diff
@@ -0,0 +1,203 @@
+- Feature Name: for-loops
+- Start Date: 2021-01-20
+
+# Summary
+[summary]: #summary
+
+This RFC gives a concrete proposal for adding `for` loops to Dafny.
+
+# Motivation
+[motivation]: #motivation
+
+Dafny has only one kind of loop, `while` loops (with either one guard or with a set of guards). Some tasks
+where other languages require a loop can be done in Dafny with the aggregate `forall` statement. Still,
+some situations call for a loop with a loop index just like what a typical `for` loop provides. This RFC
+gives a concrete proposal, so that design aspects of `for` loops in the context of Dafny can be discussed.
+
+Two central design points are:     
+- the half-open interval of values for which the loop body is performed (e.g., `0 <= i < a.Length`) and
+  the closed interval that appears in the loop invariant (e.g., `0 <= i <= a.Length`)
+- what should be the possible increments of the loop index (e.g., `+1`, `-1`, `+2`)
```

Comment:
In my opinion, increments other than `+1` and `-1` are a non-issue. It happens rarely and when it does, we simply use a `while` loop.